File: Taverne/src/app/model/comptes/compte.ts
export class Compte {
  constructor(
    protected _id?: number,
    protected _nom?: string,
    protected _prenom?: string,
    protected _login?: string,
    protected _password?: string,
    protected _mail?: string,
    protected _version?: number,
    protected _enabled?: boolean,
    private _birthday?: Date
  ) {}

  get id(): number | undefined {
    return this._id;
  }

  get nom(): string | undefined {
    return this._nom;
  }

  get prenom(): string | undefined {
    return this._prenom;
  }

  get login(): string | undefined {
    return this._login;
  }

  get password(): string | undefined {
    return this._password;
  }

  get mail(): string | undefined {
    return this._mail;
  }

  get version(): number | undefined {
    return this._version;
  }

  get enabled(): boolean | undefined {
    return this._enabled;
  }

  set id(value: number | undefined) {
    this._id = value;
  }

  set nom(value: string | undefined) {
    this._nom = value;
  }

  set prenom(value: string | undefined) {
    this._prenom = value;
  }

  set login(value: string | undefined) {
    this._login = value;
  }

  set password(value: string | undefined) {
    this._password = value;
  }

  set mail(value: string | undefined) {
    this._mail = value;
  }

  set version(value: number | undefined) {
    this._version = value;
  }

  set enabled(value: boolean | undefined) {
    this._enabled = value;
  }
  /**
   * Getter birthday
   * @return {Date}
   */
  public get birthday(): Date | undefined {
    return this._birthday;
  }

  /**
   * Setter birthday
   * @param {Date} value
   */
  public set birthday(value: Date | undefined) {
    this._birthday = value;
  }
}

File: TaverneSpringBoot/src/main/java/soprajc/TaverneSpringBoot/model/comptes/Compte.java
package soprajc.TaverneSpringBoot.model.comptes;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Version;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.fasterxml.jackson.annotation.JsonView;

import soprajc.TaverneSpringBoot.model.JsonViews;



@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="Type_Compte")
public abstract class Compte implements UserDetails{
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@JsonView(JsonViews.Common.class)
	protected Long id;
	@JsonView(JsonViews.Common.class)
	protected String nom;
	@JsonView(JsonViews.Common.class)
	protected String prenom;
	@JsonView(JsonViews.Common.class)
	@Column(unique= true, nullable=false)
	protected String login;
	@Column(nullable=false)
	protected String password;
	@JsonView(JsonViews.Common.class)
	@Column(nullable=false)
	protected String mail;
	@JsonView(JsonViews.Common.class)
	@Column(name="date_naissance",nullable=true)
	protected LocalDate birthday;
	@Version
	protected int version;
	@Column(nullable=false)
	protected boolean enabled;
	
	public Compte() {}

	public Compte(String nom, String prenom, String login, String password, String mail) {
		this.nom = nom;
		this.prenom = prenom;
		this.login = login;
		this.password = password;
		this.mail = mail;
	}

	public String getNom() {
		return nom;
	}
	
	public void setNom(String nom) {
		this.nom = nom;
	}
	
	public String getPrenom() {
		return prenom;
	}
	
	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}
	
	public String getLogin() {
		return login;
	}
	
	public void setLogin(String login) {
		this.login = login;
	}
	
	public String getPassword() {
		return password;
	}
	
	public void setPassword(String password) {
		this.password = password;
	}
	
	public String getMail() {
		return mail;
	}
	
	public void setMail(String mail) {
		this.mail = mail;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public String toString() {
		return "Compte [id=" + id + ", nom=" + nom + ", prenom=" + prenom + ", login=" + login + ", password="
				+ password + ", mail=" + mail + "]";
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Compte other = (Compte) obj;
		return Objects.equals(id, other.id);
	}
	
	@Override
	public String getUsername() {
		return login;
	}

	public boolean isEnabled() {
		return enabled;
	}

	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}

	@Override
	public boolean isAccountNonExpired() {
		return true;
	}

	@Override
	public boolean isAccountNonLocked() {
		return true;
	}

	@Override
	public boolean isCredentialsNonExpired() {
		return true;
	}
	
	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		Collection<SimpleGrantedAuthority> authorities=new ArrayList<SimpleGrantedAuthority>();
		authorities.add(new SimpleGrantedAuthority(this.getClass().getSimpleName()));
		return authorities;
	}
	

}




